xbmc/filesystem/SmartPlaylistDirectory.cpp

```diff
@@ -9,6 +9,7 @@
 #include "SmartPlaylistDirectory.h"
 
 #include "FileItem.h"
+#include "PlayListPlayer.h"
```

Comment:
sorry this doesn't look right, I understand why you include the playlist player here (because of the playlist types) but I think that for this to go forward you'll have to decouple playlist types from the playlistplayer header first

Reply:
I am not sure what would be the best header to move the definitions of these types of playlists
maybe /playlists/playlist.h ?

Reply:
that sounds like a good place to me.
Maybe take that as a separate PR because it'd be nice to also get rid of the defines moving them to an enum classes (although you'll have to make sure python is not affected).

Reply:
playlist defines are used also on python but in this case can be changed with enum class without problems with future pr